src/modules/app/__tests__/CategoriesRouteStateView.spec.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+import { CategoryModel } from '@integreat-app/integreat-api-client'
+import CategoriesRouteStateView from '../CategoriesRouteStateView'
+import moment from 'moment-timezone'
+
+describe('CategoriesRouteStateView', () => {
+  const models = {
+    '/augsburg/de': new CategoryModel({
+      root: true,
+      path: '/augsburg/de',
+      title: 'Stadt Augsburg',
+      content: 'lul',
+      parentPath: '',
+      order: 0,
+      availableLanguages: new Map(),
+      lastUpdate: moment.tz('2017-11-18 19:30:00', 'UTC')
+    }),
+    '/augsburg/de/erste-hilfe': new CategoryModel({
+      root: false,
+      path: '/augsburg/de/erste-hilfe',
+      title: 'Stadt Augsburg',
+      content: 'lul',
+      parentPath: '/augsburg/de',
+      order: 0,
+      availableLanguages: new Map(),
+      lastUpdate: moment.tz('2017-11-18 19:30:00', 'UTC')
+    }),
+    '/augsburg/de/zweite-hilfe': new CategoryModel({
+      root: false,
+      path: '/augsburg/de/zweite-hilfe',
+      title: 'Stadt Augsburg',
+      content: 'lul',
+      parentPath: '/augsburg/de/erste-hilfe',
+      order: 0,
+      availableLanguages: new Map(),
+      lastUpdate: moment.tz('2017-11-18 19:30:00', 'UTC')
+    })
+  }
+
+  const children = {
+    '/augsburg/de': ['/augsburg/de/erste-hilfe'],
+    '/augsburg/de/erste-hilfe': ['/augsburg/de/zweite-hilfe'],
+    '/augsburg/de/zweite-hilfe': []
+  }
+
+  it('should throw if accessing root while the root model is not available', () => {
+    const stateView = new CategoriesRouteStateView('/augsburg/de/erste-hilfe', {}, {})
+    expect(() => stateView.root()).toThrowError()
+  })
+
+  it('should throw if accessing children while children are not available', () => {
+    const stateView = new CategoriesRouteStateView('/augsburg/de/erste-hilfe', {}, {})
+    expect(() => stateView.children()).toThrowError()
+  })
+
+  it('should return models of children when calling children', () => {
+    const stateView = new CategoriesRouteStateView('/augsburg/de', models, children)
+    expect(stateView.children()).toEqual([models['/augsburg/de/erste-hilfe']])
+  })
+
+  it('should enable stepping into a child category', () => {
+    const stateView = new CategoriesRouteStateView('/augsburg/de', models, children)
+    const child = stateView.stepInto('/augsburg/de/erste-hilfe')
```

Comment:
Maybe it would be beneficial to test the behavior if the path (paramof stepInto) is not available.

Reply:
Hmm...  I think, `stepInto` should only be called with param `path` if the user of this method already knows, that `path` is in the subtree of the stateView (which is true).
So this use-case should not be part of the tests, but I'll add some documentation.